SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4209,3 +4215,109 @@ void CSolver::BasicLoadRestart(CGeometry *geometry, const CConfig *config, const
                    string("It could be empty lines at the end of the file."), CURRENT_FUNCTION);
   }
 }
+
+#ifdef HAVE_LIBROM
+void CSolver::SavelibROM(CSolver** solver, CGeometry *geometry, CConfig *config, bool converged) {
+  
+  unsigned long iPoint, total_index;
+  unsigned short iVar;
+  
+  bool unsteady            = (config->GetTime_Marching() != TIME_MARCHING::STEADY);
+  string filename          = config->GetlibROMbase_FileName();
+  unsigned short pod_basis = config->GetKind_PODBasis();
+  unsigned long TimeIter   = config->GetTimeIter();
+  unsigned long nTimeIter  = config->GetnTime_Iter();
+  int maxBasisDim          = config->GetMax_BasisDim();
+  int dim = int(nPointDomain * nVar);
+  bool incremental = false;
+  
+  // Get solver nodes
+  CVariable* nodes = GetNodes();
+
+  if (!u_basis_generator) {
+    
+    /*--- Define SVD basis generator ---*/
+    int timesteps = (int)(nTimeIter - TimeIter);
+    CAROM::Options svd_options = CAROM::Options(dim, timesteps, -1,
+                                                false, true).setMaxBasisDimension(int(maxBasisDim));
+    
+    if (pod_basis == STATIC_POD) {
+      std::cout << "Creating static basis generator." << std::endl;
+      if (unsteady) {
+        std::cout << "Incremental basis generator recommended for unsteady simulations." << std::endl;
```

Comment:
Your messages will be printed multiple times in MPI runs, and you will have multiple "basis generators", is that ok? Or are you looking for a better integration with MPI, we can help with that.

Reply:
Oh yes, I meant for the messages to only be printed once. I updated those with (rank == MASTER_NODE) if statements. The multiple basis generators is intentional